lib/resource/changes/create_new_version.ex

```diff
@@ -21,26 +18,85 @@ defmodule AshPaperTrail.Resource.Changes.CreateNewVersion do
   def atomic(changeset, opts, context) do
     change_tracking_mode = AshPaperTrail.Resource.Info.change_tracking_mode(changeset.resource)
 
+    has_after_batch_hooks? =
```

Comment:
I think if you implement the `after_batch` callback we don't need the `atomic` callback at all, since all of the work will happen there?

Reply:
You are right. I updated the logic of the atomic callback accordingly.